### PR TITLE
Provide the way to disable certain dates [dxCalendar, dxDateBox]

### PR DESCRIPTION
**Pull request by DokaRus**

### CODE REVIEW

**Review by RomanTsukanov — DISMISSED** [review]

**Review by mrjohn0011 — DISMISSED** [review]

**Review by vconst — DISMISSED** [review]

File: js/ui/calendar/ui.calendar.js [hunk 1]
```diff
@@ -283,12 +301,15 @@ var Calendar = Editor.inherit({
                     ? max
                     : dateUtils.getViewLastCellDate(zoomLevel, currentDate);
 
-                this.option("currentDate", date);
+                if(this._view.isDateDisabled(date)) {
+                    this._moveCurrentDate(-1, date);
+                } else {
+                    this.option("currentDate", date);
+                }
```
Comment (mrjohn0011) [inline js/ui/calendar/ui.calendar.js]: Extract a method please

File: testing/tests/DevExpress.ui.widgets.editors/calendarViews.tests.js [hunk 2]
```diff
@@ -662,6 +662,105 @@ QUnit.test("monthView should not allow to navigate to a date earlier than min an
...
+    assert.ok(disabledDays.indexOf(this.view.option("value").getDate()));
+});
+
+QUnit.test("monthView should not allow to navigate to a disabled date", function(assert) {
+    var $element = this.$element,
+        view = this.view,
+        trigger = function(which) { var e = $.Event("keydown", { which: which }); $element.find("table").trigger(e); };
+
+    view.option("contouredDate", new Date(2010, 10, 5));
+    trigger(38);
```
Comment (mrjohn0011) [inline testing/tests/DevExpress.ui.widgets.editors/calendarViews.tests.js]: define a constant please

File: ts/widgets-base.d.ts [hunk 3]
```diff
@@ -1863,6 +1863,9 @@ declare module DevExpress.ui {
         /** @docid dxDateBoxOptions_interval */
         interval?: number;
 
+        /** @docid dxDateBoxOptions_disabledDates */
+        disabledDates?: any;
```
Comment (mrjohn0011) [inline ts/widgets-base.d.ts:1867]: discuss with MVC team about Array<date> type

Comment (DokaRus) [reply ts/widgets-base.d.ts:1867]: We'll choose to stay with the current TS definitions

**Review by mrjohn0011 — APPROVED** [review]

File: ts/widgets-base.d.ts (hunk 3 above)
Comment (mrjohn0011) [reply ts/widgets-base.d.ts:1867]: ok

**Review by lukyanovas — APPROVED** [review]

**Pull request merged**